### PR TITLE
[MACROS][SET_GERMAN_AND_PUBLISH] synchronise added package when Engli…

### PR DESCRIPTION
**Pull request by jerome-poisson**
…sh item is corrected

SDFID-606

### CODE REVIEW

**Comment by jerome-poisson** [comment]
This PR is nearly finished, but I have still an error to investigate when removing an item after adding one. Review is welcome though

**Comment by jerome-poisson** [comment]
**note**: a PR will follow for core, as metadata are not set correctly when correcting a package and adding a item in it.

File: server/fidelity/macros/set_german_and_publish.py [hunk 1]
```diff
@@ -99,6 +120,119 @@ def set_german_and_publish(item, **kwargs):
...
+                    de_group['refs'].append(new_ref)
+
+                # now we remove items removed from English version
+                deleted_refs = en_de_map.keys() - ori_refs
+                for deleted_ref in deleted_refs:
+                    de_deleted_ref = en_de_map[deleted_ref]
+                    for ref in de_group['refs']:
+                        if ref.get('guid') == de_deleted_ref:
+                            break
+                    de_group['refs'].remove(ref)
```
Comment (ride90) [inline server/fidelity/macros/set_german_and_publish.py:234]: shouldn't it be
```
for ref in de_group['refs']:
    if ref.get('guid') == de_deleted_ref:
        break
    de_group['refs'].remove(ref)
```
?

Comment (jerome-poisson) [reply server/fidelity/macros/set_german_and_publish.py:234]: no this is fine, it must be removed when it's matching the `de_delete_ref`. To make it more explicit, I've moved it above the `break` now (will push soon).

File: server/fidelity/macros/set_german_and_publish.py [hunk 2]
```diff
@@ -99,6 +120,119 @@ def set_german_and_publish(item, **kwargs):
...
+                                    'missing "processed_from" key in referenced item {ref_item_id} from package '
+                                    '{item_id}'.format(
+                                        ref_item_id=ref_item_id,
+                                        item_id=ref_item[config.ID_FIELD]))
+                                continue
+
+            # we delete groups in German package which don't exist in English one
+            groups_to_delete = []
+            for deleted_group in dup_group_en_de_map.keys() - ori_group_refs.keys():
+                for group in groups:
```
[the remark this answers is not in the document]
Comment (jerome-poisson) [reply server/fidelity/macros/set_german_and_publish.py:161]: right, it's created above but in a different path, I've moved the `groups` creation above so it's available, thanks!

**Review by ride90 — APPROVED** [review]

**Pull request merged**